### ISSUE TITLE
prefer-const false positive in for loop

### ISSUE THREAD

**Issue by Avol-V** [palantir/tslint#2195]
### Bug Report

- __TSLint version__: `4.4.2`
- __TypeScript version__: `2.1.6`
- __Running TSLint via__: Visual Studio Code

#### TypeScript code being linted

```ts
for ( let i = 0, n = 10; i < n; i++ )
{
	console.log( i );
}
```

with `tslint.json` configuration:

```json
"prefer-const": true,
```

#### Actual behavior

```
[tslint] Identifier 'n' is never reassigned; use 'const' instead of 'let'. (prefer-const)
```

#### Expected behavior

I can't do this in `for` loop.

**Comment by nchen63** [comment]
can you pull out `n` from the loop?

**Comment by Avol-V** [comment]
It's just a simplified example but with common using of `for`, for my opinion. The main idea there is that I need this variables to be only inside loop block, and I can do this by defining it in initialisation section of `for`. If I create my variable outside the loop, it will be available in whole block later. If I create my variable inside the loop, it will be created on every iteration.

**Comment by ajafff** [comment]
I get your point. This also bugged me several times.
A real world example:
```ts
for (let i = 1, l = arr.length; i<l; ++i)
    // do stuff
```

To summarize: this is more a feature request than a bug report.
The request is for an option to ignore VariableDeclarationLists where at least one variable is reassigned.
Am I right @Avol-V?

**Comment by Avol-V** [comment]
@ajafff yes it can be an option for any list, but I think it much better to ignore just `for` loops, like in `one-variable-per-declaration` rule.

**Comment by ajafff** [comment]
I'm currently working on a rewrite of this rule and will make it configurable while I'm already there. You can expect a PR this week.
